buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt

```diff
@@ -201,7 +194,10 @@ open class ShadedJarPlugin : Plugin<Project> {
                 attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
             }
             extendsFrom(shadedImplementation)
-            outgoing.artifact(shadedJarArtifact)
+            outgoing.artifact(shadedJarTask) {
+                name = base.archivesBaseName
```

Comment:
:o: Why do we do this only at publication time?

Reply:
Because the `ShadedJarTask` is not a Jar/Archive task but our own task implemetnation extending DefaultTask. So the special handling of archive tasks does not work.
That should probably be changed in the future (I don't know why it was done that way)

Reply:
That task should also be abstract so we don't need the `project.objects.fileCollection()` boilerplate etc. in there. I'll look into that in a follow up.